### PR TITLE
BUG: Add gcd/lcm definitions to npy_math.h

### PR DESCRIPTION
**Pull request by sethtroisi**
gcd/lcm was added in https://github.com/numpy/numpy/commit/7ab0f15250f5965a1e47e419597ca358e9b31a75

I believe it was an oversight to not add the function definitions to npy_math.h, which resulted in an **implicit function declaration** for me which means the result of gcd is truncated to 32 bits (which is correct for the vast majority of results) but not all.

Luckily the [gcd_overflow test](https://github.com/numpy/numpy/blob/master/numpy/core/tests/test_umath.py#L2480) caught this. Unfortantely I had **-w** on and didn't see the **-Wimplicit-function-declaration** error and the test case result is 2^64 which was truncated to 0 so it took quite a bit of my time to debug. 

I'm not sure if this should be labelled BLD, ENH, or MAINT. 
I'm also not sure where in the file you'd like these sorted.

### CODE REVIEW

**Comment by eric-wieser** [comment]
> which means the result of gcd is truncated to 32 bits

Sounds like "BUG:" to me

**Comment by eric-wieser** [comment]
> Luckily the gcd_overflow test caught this. 

Why did it do so on your machine but not CI?

**Review by eric-wieser — APPROVED** [review]

**Comment by seberg** [comment]
My guess is this would have to be added to `numpy/core/include/numpy/npy_math.h` and the reason is that the `NPY_INLINE_MATH` is almost always True. That in turn gets toggled by `NPY_INTERNAL_BUILD`, which should always be True while building numpy?

Cannot say I know the building/compiling that well, but is there anything not quite typical about your setup/compiler?!

**Comment by sethtroisi** [comment]
@seberg I think you're correct, I don't have NPY_INLINE_MATH turned on (not sure why but not something I'm going to poke today).

@eric-wieser I'm not running numpy in a normal config (I'm building with BAZEL) which is why this is happening for me but not in CI, see @seberg's analysis on why this probably happens.

**Comment by seberg** [comment]
So I guess the issue is really with BAZEL here. If you wish to followup we can just update here and change the title. Otherwise I might move it to an issue at some point.

**Comment by sethtroisi** [comment]
Thanks for comments,

I might not get to this in the next two days (personal life...) but I'll have an update Monday.

I do believe these should be included in npy_math.h, as they are used in loops.c.src which I believe uses them implicitly.

File: numpy/core/include/numpy/npy_math.h [hunk 1]
```diff
@@ -145,6 +145,24 @@ NPY_INLINE static float __npy_nzerof(void)
...
+npy_ulong npy_lcmul(npy_ulong a, npy_ulong b);
+npy_ulonglong npy_gcdull(npy_ulonglong a, npy_ulonglong b);
+npy_ulonglong npy_lcmull(npy_ulonglong a, npy_ulonglong b);
+
+npy_int npy_gcd(npy_int a, npy_int b);
+npy_int npy_lcm(npy_int a, npy_int b);
+npy_long npy_gcdl(npy_long a, npy_long b);
+npy_long npy_lcml(npy_long a, npy_long b);
+npy_longlong npy_gcdll(npy_longlong a, npy_longlong b);
+npy_longlong npy_lcmll(npy_longlong a, npy_longlong b);
```
Comment (eric-wieser) [inline numpy/core/include/numpy/npy_math.h]: needs `NPY_INPLACE` like all the other functions in this file.

Comment (sethtroisi) [reply numpy/core/include/numpy/npy_math.h]: Thanks!

**Comment by seberg** [comment]
I guess this is the correct fix then, do you want to just merge it Eric?

**Comment by eric-wieser** [comment]
It's a shame our CI didn't catch this, but the change looks very safe to me.

**Pull request merged**